bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -31,14 +39,65 @@ public void apply(final ClassModelBuilder<?> classModelBuilder) {
             processClassAnnotation(classModelBuilder, annotation);
         }
 
-        for (final FieldModelBuilder<?> fieldModelBuilder : classModelBuilder.getFields()) {
-            for (final Annotation annotation : fieldModelBuilder.getAnnotations()) {
-                processFieldAnnotation(classModelBuilder, fieldModelBuilder, annotation);
+        for (final PropertyModelBuilder<?> propertyModelBuilder : classModelBuilder.getPropertyModelBuilders()) {
+            for (final Annotation annotation : propertyModelBuilder.getAnnotations()) {
+                processPropertyAnnotation(classModelBuilder, propertyModelBuilder, annotation);
+            }
+        }
+
+        processCreatorAnnotation(classModelBuilder);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> void processCreatorAnnotation(final ClassModelBuilder<T> classModelBuilder) {
+        Class<T> clazz = classModelBuilder.getType();
+        CreatorExecutable<T> creatorExecutable = null;
+        for (Constructor<?> constructor : clazz.getDeclaredConstructors()) {
+            if (isPublic(constructor.getModifiers())) {
+                for (Annotation annotation : constructor.getDeclaredAnnotations()) {
+                    if (annotation.annotationType().equals(Creator.class)) {
+                        creatorExecutable = new CreatorExecutable<T>(clazz, (Constructor<T>) constructor);
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (creatorExecutable == null) {
+            for (Method method : clazz.getDeclaredMethods()) {
+                if (isStatic(method.getModifiers())) {
+                    for (Annotation annotation : method.getDeclaredAnnotations()) {
+                        if (annotation.annotationType().equals(Creator.class)) {
+                            creatorExecutable = new CreatorExecutable<T>(clazz, method);
+                            break;
+                        }
+                    }
+                }
             }
         }
+
+        if (creatorExecutable != null) {
+            List<Property> properties = creatorExecutable.getProperties();
+            List<Class<?>> parameterTypes = creatorExecutable.getParameterTypes();
+            if (properties.size() != parameterTypes.size()) {
+                throw creatorExecutable.getError("All parameters must be annotated with a @Property");
+            }
+            for (int i = 0; i < properties.size(); i++) {
+                Property property = properties.get(i);
+                Class<?> parameterType = parameterTypes.get(i);
+                PropertyModelBuilder<?> propertyModelBuilder = classModelBuilder.getProperty(property.value());
+                if (propertyModelBuilder == null) {
+                    throw creatorExecutable.getError(format("Missing Property with the value: '%s'", property.value()));
```

Comment:
In Yasson this isn't actually an error.  See org.eclipse.yasson.customization.JsonbCreatorTest#testCreatorWithoutJavabeanProperty

In other words it's perfectly ok to use a creator with a property defined by one of its arguments that doesn't match up with any property defined in the model.